src/backend/distributed/deparser/deparse_function_stmts.c

```diff
@@ -0,0 +1,527 @@
+static void AppendDropFunctionStmt(StringInfo buf, DropStmt *stmt);
+static void AppendFunctionName(StringInfo buf, ObjectWithArgs *func);
+static void AppendFunctionNameList(StringInfo buf, List *objects);
+
+static void AppendDefElem(StringInfo buf, DefElem *def);
+static void AppendDefElemStrict(StringInfo buf, DefElem *def);
+static void AppendDefElemVolatility(StringInfo buf, DefElem *def);
+static void AppendDefElemLeakproof(StringInfo buf, DefElem *def);
+static void AppendDefElemSecurity(StringInfo buf, DefElem *def);
+static void AppendDefElemParallel(StringInfo buf, DefElem *def);
+static void AppendDefElemCost(StringInfo buf, DefElem *def);
+static void AppendDefElemRows(StringInfo buf, DefElem *def);
+static void AppendDefElemSet(StringInfo buf, DefElem *def);
+
+static void AppendRenameFunctionStmt(StringInfo buf, RenameStmt *stmt);
+static void AppendAlterFunctionSchemaStmt(StringInfo buf, AlterObjectSchemaStmt *stmt);
+static void AppendAlterFunctionOwnerStmt(StringInfo buf, AlterOwnerStmt *stmt);
+static void AppendAlterFunctionDependsStmt(StringInfo buf, AlterObjectDependsStmt *stmt);
+
+
+const char *
+DeparseAlterFunctionStmt(AlterFunctionStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	AppendAlterFunctionStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+const char *
+DeparseRenameFunctionStmt(RenameStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
+	Assert(stmt->renameType == OBJECT_FUNCTION);
+#else
+	Assert(stmt->renameType == OBJECT_FUNCTION || stmt->renameType == OBJECT_PROCEDURE);
+#endif
+
+	AppendRenameFunctionStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+const char *
+DeparseAlterFunctionSchemaStmt(AlterObjectSchemaStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
+	Assert(stmt->objectType == OBJECT_FUNCTION);
+#else
+	Assert(stmt->objectType == OBJECT_FUNCTION || stmt->objectType == OBJECT_PROCEDURE);
+#endif
+
+	AppendAlterFunctionSchemaStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+const char *
+DeparseAlterFunctionOwnerStmt(AlterOwnerStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
+	Assert(stmt->objectType == OBJECT_FUNCTION);
+#else
+	Assert(stmt->objectType == OBJECT_FUNCTION || stmt->objectType == OBJECT_PROCEDURE);
+#endif
+
+	AppendAlterFunctionOwnerStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+const char *
+DeparseAlterFunctionDependsStmt(AlterObjectDependsStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
+	Assert(stmt->objectType == OBJECT_FUNCTION);
+#else
+	Assert(stmt->objectType == OBJECT_FUNCTION || stmt->objectType == OBJECT_PROCEDURE);
+#endif
+
+	AppendAlterFunctionDependsStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+const char *
+DeparseDropFunctionStmt(DropStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
```

Comment:
Do we need these asserts in `Deparse*()` functions? We're only getting here if these conditions hit and we're already guarding every deparse with a test. 

These are very repetitive and ifdefs make it ugly :)

Reply:
I will create a new commit to drop all these assertions in `Deparse*()` functions. I guess I will still need to test that all the callers already make sure that these conditions are hit.

Reply:
You were right, and there is no harm in dropping these assertions. Thanks for this catch